Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -817,15 +841,27 @@ internal struct TextFormatScanner {
             throw TextFormatDecodingError.malformedText
         }
         let c = p[0]
+        p += 1
         switch c {
-        case asciiZero, asciiOne, asciiLowerF, asciiUpperF, asciiLowerT, asciiUpperT:
-            switch parseIdentifier() {
-            case "0", "f", "false", "False":
+        case asciiZero:
+            return false
+        case asciiOne:
+            return true
+        case asciiLowerF, asciiUpperF:
+            let alse = [asciiLowerA, asciiLowerL, asciiLowerS, asciiLowerE]
+            if p == end
+               || skipOptionalKeywordCaseSensitive(bytes: alse)
+               || ((p[0] < asciiUpperA || p[0] > asciiUpperZ)
+                   && (p[0] < asciiLowerA || p[0] > asciiLowerZ)) {
                 return false
-            case "1", "t", "true", "True":
+            }
+        case asciiLowerT, asciiUpperT:
+            let rue = [asciiLowerR, asciiLowerU, asciiLowerE]
+            if p == end
+               || skipOptionalKeywordCaseSensitive(bytes: rue)
+               || ((p[0] < asciiUpperA || p[0] > asciiUpperZ)
```

Comment:
Same concern.

Seems like in all these cases, you might want to set a proposed value, and then ensure the next thing is whitespace or a TextFormat delimiter to the malformed inputs ("0foo:1", etc.), don't extract a subset and then parse the rest of this token as a field.

Reply:
Fixed (and tested).